File: Formula-Bollo-App/src/main/java/formula/bollo/app/entity/Penalty.java
package formula.bollo.app.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

import lombok.Data;

@Entity
@Table(name = "penalty")
@Data
public class Penalty {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "race_id", referencedColumnName = "id")
    private Race race;

    @ManyToOne
    @JoinColumn(name = "driver_id", referencedColumnName = "id")
    private Driver driver;

    @Column(name = "reason", length = 250)
    private String reason;

    @ManyToOne
    @JoinColumn(name = "severity_id", referencedColumnName = "id")
    private PenaltySeverity severity;

    @ManyToOne
    @JoinColumn(name = "season_id", referencedColumnName = "id")
    private Season season;
}

File: Formula-Bollo/src/shared/models/penalty.ts
import { Driver } from "./driver";
import { PenaltySeverity } from "./penaltySeverity";
import { Race } from "./race";
import { Season } from "./season";

export class Penalty {
  id: number;
  race: Race;
  driver: Driver;
  severity: PenaltySeverity;
  reason: string;
  season: Season | undefined;

  constructor(
    id: number,
    race: Race,
    driver: Driver,
    severity: PenaltySeverity,
    reason: string,
    season?: Season,
  ) {
    this.id = id;
    this.race = race;
    this.driver = driver;
    this.severity = severity;
    this.reason = reason;
    this.season = season;
  }
}
